rfcs/build-system/04-build-shield.md

```diff
@@ -0,0 +1,107 @@
+
+#### Keeping builds unblocked
+
+This includes monitoring for:
+
+- Automated failure notifications in "Build failures (automated)" channel
+- Messages from team members (mainly in "Engineering Systems" channel) about PR or local build issues
+- Messages or github mentions from regular shield about urgent build-related regressions/breaks that showed up as github issues
+  - This is much less common and will primarily be up to the regular shield PM/dev to watch for these issues and loop in build shield
+
+Prioritization is roughly as follows. As with normal shield, it's fine to pull in others for consultation/help as needed, especially if it's a particularly complex or urgent issue, or if a more in-depth fix turns out to be necessary.
+
+| Issue type                                      | Priority                                                                     |
+| ----------------------------------------------- | ---------------------------------------------------------------------------- |
+| Release build failed                            | 🔥 Investigate ASAP (fix issue, or retry if intermittent)                    |
+| CI _consistently_ broken                        | 🔥 Fix ASAP                                                                  |
+| PR builds _consistently_ broken (CI okay)       | 🔥 Fix ASAP                                                                  |
+| Local builds _consistently_ broken (PR/CI okay) | 🔥 Fix ASAP                                                                  |
+| Published package broken                        | 🔥 Fix ASAP (usually due to dep issue or missing file)                       |
+| CI _intermittently_ broken                      | ⚠️ Try to investigate/fix, or file an issue if it will take significant time |
+| v7/8 website release failed (uncommon)          | ⚠️ Re-run failed stage, or see internal wiki                                 |
+
+#### Documenting issues and troubleshooting procedures
+
+The second responsibility is **documenting** failures and troubleshooting procedures, with the goals of:
+
+- Noticing failure patterns which we need to schedule work to address
+- Establishing a "runbook" for troubleshooting build issues
+  - The goal of this is to de-mystify build, and maybe even (eventually) be able to expand the build shield rotation to more team members, so that everyone has exposure to the area
+
+Where/how to document:
+
+- Troubleshooting guides should go in the internal wiki
+- Comment on automated failure posts with the cause of the failure and the resolution
+  - We may also want to make a form or spreadsheet or something to more easily gather metrics on failures
+
+### Non-responsibilities (initially)
+
+At least initially, build shield is NOT responsible for general build improvements/maintenance, such as:
```

Comment:
In general I see where you coming from and kinda agree with this RFC although from my experience "just 25%" allocation and "firefighting" is a myth and in general a dedicated team is needed to keep moving us forward in quick/effective way so feature teams (components) can focus solely on their delivery - aka those teams/devs are clients of tooling/build teams. with that the support and troubleshooting is spread across other folks as people responsible for tooling have the time to explain and educate others in proper fashion.

Reply:
I agree, if we can get the effort onboard for this it would be awesome

Reply:
I agree about getting ongoing efforts scheduled for improvements (and maybe the "build shield" responsibility could be combined with that), but unless we can start that right away, this is an incremental improvement in the meantime.

Fundamentally the problem is **if the build breaks, who is responsible to investigate?** Even with a better process for overall improvements, random things will break from time to time. And even if we had designated ownership areas among the build team (which it seems like might be is part of what you're describing? if not ignore this), someone needs to be on point for initial investigation. 

This responsibility needs to be better distributed among build team members, and better documented so newer build team members can learn/start helping.

Reply:
Oh also, scheduling ongoing improvements won't necessarily solve the time zone issue. Like if someone from Prague is the only one assigned to that work in a given month, who's on point to investigate breaks and keep people unblocked on Redmond time? (By default that's been me, but we need to spread that out.)